embassy-stm32/src/i2c/v2.rs

```diff
@@ -382,35 +383,59 @@ impl<'d, T: Instance, TXDMA, RXDMA> I2c<'d, T, TXDMA, RXDMA> {
         // I2C start
         //
         // ST SAD+W
-        Self::master_write(
+        match Self::master_write(
```

Comment:
please write these with "early return" style. This avoids too much neested indentation and keeps the error handling close to the source errors.


```rust
if let Err(e) = Self::master_write(..) {
   ... handle it
   return Err(e)
}